python/tank/deploy/tank_commands/core_upgrade.py

```diff
@@ -251,184 +256,133 @@ def __init__(self, installation_root, logger, core_version=None):
                                   e.g. you can run the updater as a totally separate thing.
         :param logger: Logger to send output to.
         :param core_version: Version of the core to update to. If None, the core will be updated to the latest
-                             version.
+                             version. Defaults to None.
```

Comment:
Actually, i think we are overcomplicating things by calling this `core_constraint_pattern` - that implies a `v1.x.x` to me - is there a use case when we need this? Can't it just be a plain version number for now? That simplifies things a fair bit i think. We don't do a proper semantic versioning for core right now so the constraints pattern doesn't really make sense at this point I think - it kind of assumes a stricter treatment of version increments than we are currently doing in core.